packages/vite/src/node/plugins/define.ts

```diff
@@ -89,7 +91,8 @@ export function definePlugin(config: ResolvedConfig): Plugin {
       }
 
       if (
-        // exclude css and static assets for performance
+        // exclude html, css and static assets for performance
+        isHTMLRequest(id) ||
```

Comment:
IIUC, the code will only run in build or ssr mode, so we test it in build, right?

Reply:
Oh, wait... my mistake, misunderstanding the intent of the code, we are preventing defining HTML code in dev . 😅
Sorry for the noise